overlord/devicestate/firstboot_test.go

```diff
@@ -171,56 +179,136 @@ func (s *FirstBootTestSuite) TestPopulateFromSeedErrorsOnState(c *C) {
+	info, err := snap.InfoFromSnapYaml([]byte(snapYaml))
 	c.Assert(err, IsNil)
+	snapName := info.Name()
 
-	// put a firstboot local snap into the SnapBlobDir
-	snapYaml = `name: local
-version: 1.0`
-	mockSnapFile = snaptest.MakeTestSnapWithFiles(c, snapYaml, nil)
-	targetSnapFile2 := filepath.Join(dirs.SnapSeedDir, "snaps", filepath.Base(mockSnapFile))
-	err = os.Rename(mockSnapFile, targetSnapFile2)
-	c.Assert(err, IsNil)
+	mockSnapFile := snaptest.MakeTestSnapWithFiles(c, snapYaml, files)
+	snapFname = filepath.Base(mockSnapFile)
 
-	devAcct := assertstest.NewAccount(s.storeSigning, "developer", map[string]interface{}{
-		"account-id": "developerid",
-	}, "")
-	devAcctFn := filepath.Join(dirs.SnapSeedDir, "assertions", "developer.account")
-	err = ioutil.WriteFile(devAcctFn, asserts.Encode(devAcct), 0644)
+	targetFile := filepath.Join(dirs.SnapSeedDir, "snaps", snapFname)
+	err = os.Rename(mockSnapFile, targetFile)
 	c.Assert(err, IsNil)
 
-	snapDecl, err := s.storeSigning.Sign(asserts.SnapDeclarationType, map[string]interface{}{
+	declA, err := s.storeSigning.Sign(asserts.SnapDeclarationType, map[string]interface{}{
 		"series":       "16",
-		"snap-id":      "snapidsnapid",
-		"publisher-id": "developerid",
-		"snap-name":    "foo",
+		"snap-id":      snapName + "-snap-id",
+		"publisher-id": developerID,
+		"snap-name":    snapName,
 		"timestamp":    time.Now().UTC().Format(time.RFC3339),
 	}, nil, "")
 	c.Assert(err, IsNil)
-	declFn := filepath.Join(dirs.SnapSeedDir, "assertions", "foo.snap-declaration")
-	err = ioutil.WriteFile(declFn, asserts.Encode(snapDecl), 0644)
-	c.Assert(err, IsNil)
 
-	sha3_384, size, err := asserts.SnapFileSHA3_384(targetSnapFile)
+	sha3_384, size, err := asserts.SnapFileSHA3_384(targetFile)
 	c.Assert(err, IsNil)
 
-	snapRev, err := s.storeSigning.Sign(asserts.SnapRevisionType, map[string]interface{}{
+	revA, err := s.storeSigning.Sign(asserts.SnapRevisionType, map[string]interface{}{
 		"snap-sha3-384": sha3_384,
 		"snap-size":     fmt.Sprintf("%d", size),
-		"snap-id":       "snapidsnapid",
-		"developer-id":  "developerid",
-		"snap-revision": "128",
+		"snap-id":       snapName + "-snap-id",
+		"developer-id":  developerID,
+		"snap-revision": fmt.Sprintf("%d", revno),
 		"timestamp":     time.Now().UTC().Format(time.RFC3339),
 	}, nil, "")
 	c.Assert(err, IsNil)
+
+	return snapFname, declA.(*asserts.SnapDeclaration), revA.(*asserts.SnapRevision)
+}
+
+func (s *FirstBootTestSuite) makeCoreSnaps(c *C, withConfigure bool) (coreFname, kernelFname, gadgetFname string) {
+	files := [][]string{}
+	if withConfigure {
+		files = [][]string{{"meta/hooks/configure", ""}}
+	}
+
+	// put core snap into the SnapBlobDir
+	snapYaml := `name: core
+version: 1.0
+type: os`
+	coreFname, coreDecl, coreRev := s.makeAssertedSnap(c, snapYaml, files, 1, "canonical")
```

Comment:
(nitpick^2, feel free to ignore): if we could use `s.makeAssertedSnap(..., snap.R(1), ...)` here it would feel even more readable to me.